tests/api/preprocessing/test_normalization.py

```diff
@@ -28,14 +29,14 @@ def setup_method(self):
             "Feature": ["Integer1", "Numeric1", "Numeric2", "Numeric3", "String1", "String2"],
             "Type": ["Integer", "Numeric", "Numeric", "Numeric", "String", "String"],
         }
-
         self.adata = AnnData(
             X=X_data,
             obs=pd.DataFrame(data=obs_data),
             var=pd.DataFrame(data=var_data, index=var_data["Feature"]),
             dtype=np.dtype(object),
+            uns=OrderedDict(),
         )
-
+        self.adata.uns["numerical_columns"] = ["Numeric1", "Numeric2"]
```

Comment:
Did not changed it here but to be honest: `Integer1` seems like its a numerical column and will be detected by the new function (but not from autodetect). This is due to the fact, that the sample size here are only 3 observations (unlikely to see such small sets in real world) and autodetect flagged `Integer1` as categorical, that does not need any encoding (since it has numerical dtype), but it only has like 2 different values out of 3, so autodetect flags this as categorical.

That's one of the "potential" breakings I could see (altough not a severe one and technically, this version now is the correct one).